sotodlib/tod_ops/flags.py

```diff
@@ -897,3 +916,176 @@ def get_stats(aman, signal, stat_names, split_subscans=False, mask=None, name="s
 
     info_aman = wrap_stats(aman, name, stats_arr, stat_names, merge)
     return info_aman
+
+def get_focalplane_flags(aman, merge=True, overwrite=True, invalid_flags_name='fp_flags'):
+    """
+    Generate flags for invalid detectors in the focal plane.
+        The tod.
+    merge : bool
+        If true, merges the generated flag into aman.
+    overwrite : bool
+        If true, write over flag. If false, don't.
+    invalid_flags_name : str
+        Name of flag to add to aman.flags if merge is True.
+
+    Returns
+    -------
+    msk_invalid_fp : RangesMatrix
+        RangesMatrix of invalid detectors in the focal plane.
+    """
+    # Available detectors in focalplane
+    xi_nan = np.isnan(aman.focal_plane.xi)
+    eta_nan = np.isnan(aman.focal_plane.eta)
+    x = Ranges(aman.samps.count)
+    msk_invalid_fp = RangesMatrix([
+        Ranges.ones_like(x) if Y else Ranges.zeros_like(x) 
+        for Y in flag_invalid_fp
+    ])
+    flag_valid_fp = np.sum([xi_nan, eta_nan, gamma_nan], axis=0) == 0
+    flag_invalid_fp = ~flag_valid_fp
+    msk_invalid_fp = RangesMatrix([Ranges.ones_like(x) if Y else Ranges.zeros_like(x) for Y in flag_invalid_fp])
+    
+    if merge:
+        if invalid_flags_name in aman.flags and not overwrite:
+            raise ValueError(f"Flag name {invalid_flags_name} already exists in aman.flags")
+        if invalid_flags_name in aman.flags:
+            aman.flags[invalid_flags_name] = msk_invalid_fp
+        else:
+            aman.flags.wrap(invalid_flags_name, msk_invalid_fp, [(0, 'dets'), (1, 'samps')])
+
+    return msk_invalid_fp
+
+
+def get_badsubscan_flags(aman, nstd_threshold=3.0, Tptp_pW_threshold=0.5, kurt_threshold=0.5, 
```

Comment:
Also likely we can use some of @earosenberg 's recent additions here.

Reply:
As above -- can remove the "get_subscan_flags" and we can include this in the next PR as I was only using this in the AZSS subtraction. Let me know if you have suggestions, otherwise I'll just remove it.